pkg/apb/client.go

```diff
@@ -174,20 +149,69 @@ func (c *Client) RunImage(
 	c.log.Debug("action:[ %s ]", action)
 	c.log.Debug("params:[ %s ]", string(params))
 
+	//TODO(rhallisey): The Kubernetes API objects should be in JSON structs
+	// so it's easier to manipulate them. This is for a rebase so we don't
+	// block the service account work.  I'll come back after and clean this
+	// up.
+	var pod *v1.Pod
 	if clusterConfig.InCluster {
-		return RunCommand("oc", "run", fmt.Sprintf("aa-%s", uuid.New()),
-			fmt.Sprintf("--image-pull-policy=Always"),
-			fmt.Sprintf("--image=%s", spec.Image), "--restart=Never",
-			"--", action, "--extra-vars", string(params))
+		pod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("aa-%s", uuid.New()),
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "apb",
+						Image: spec.Image,
+						Args: []string{
+							action,
+							"--extra-vars",
+							string(params),
+						},
+						ImagePullPolicy: v1.PullAlways,
+					},
+				},
+				RestartPolicy: v1.RestartPolicyNever,
+			},
+		}
 	} else {
-		return RunCommand("oc", "run", fmt.Sprintf("aa-%s", uuid.New()),
-			"--env", fmt.Sprintf("OPENSHIFT_TARGET=%s", clusterConfig.Target),
-			"--env", fmt.Sprintf("OPENSHIFT_USER=%s", clusterConfig.User),
-			"--env", fmt.Sprintf("OPENSHIFT_PASS=%s", clusterConfig.Password),
-			fmt.Sprintf("--image-pull-policy=Always"),
-			fmt.Sprintf("--image=%s", spec.Image), "--restart=Never",
-			"--", action, "--extra-vars", string(params))
+		pod = &v1.Pod{
```

Comment:
I'm concerned about the differentiation of running in cluster or out, feels wrong to need to make this choice in the code.